src/tests/quantuminspire/qiskit/test_measurements.py

```diff
@@ -0,0 +1,220 @@
+limitations under the License.
+
+"""
+import unittest
+from unittest.mock import Mock
+
+from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
+from qiskit.compiler import assemble
+from quantuminspire.qiskit.backend_qx import QuantumInspireBackend
+from quantuminspire.exceptions import QiskitBackendError
+from quantuminspire.qiskit.measurements import Measurements
+
+
+class TestMeasurements(unittest.TestCase):
+
+    def setUp(self):
+        self.measurements = Measurements()
+
+    @staticmethod
+    def _circuit_to_qobj(circuit):
+        run_config_dict = {'shots': 25, 'memory': True}
+        backend = QuantumInspireBackend(Mock(), Mock())
+        qobj = assemble(circuit, backend, **run_config_dict)
+        return qobj
+
+    @staticmethod
+    def _circuit_to_experiment(circuit):
+        qobj = TestMeasurements._circuit_to_qobj(circuit)
+        return qobj.experiments[0]
+
+    def test_collect_measurements(self):
+        qc = QuantumCircuit(2, 2)
+        qc.cx(0, 1)
+        qc.measure(0, 1)
+        qc.measure(1, 0)
+
+        experiment = self._circuit_to_experiment(qc)
+        self.measurements.collect_measurements(experiment)
+        expected_result = {'measurements_state': [[1, 0], [0, 1]],
+                           'measurements_reg': [[0, 1], [1, 0]],
+                           'number_of_qubits': 2,
+                           'number_of_clbits': 2
+                           }
+        self.assertDictEqual(self.measurements.to_dict(), expected_result)
+        self.assertEqual(self.measurements.nr_of_qubits, 2)
+        self.assertEqual(self.measurements.nr_of_clbits, 2)
+
+    def test_collect_measurements_without_measurements(self):
+        qc = QuantumCircuit(2, 2)
+        qc.cx(0, 1)
+
+        experiment = self._circuit_to_experiment(qc)
+
+        self.measurements.collect_measurements(experiment)
+        expected_result = {'measurements_state': [[0, 0], [1, 1]],
+                           'measurements_reg': [[0, 0], [1, 1]],
+                           'number_of_qubits': 2,
+                           'number_of_clbits': 2
+                           }
+        self.assertDictEqual(self.measurements.to_dict(), expected_result)
+        self.assertEqual(self.measurements.nr_of_qubits, 2)
+        self.assertEqual(self.measurements.nr_of_clbits, 2)
+
+    def test_validate_nr_classical_qubits_less_than_needed_for_storing_measured_qubits(self):
+        q = QuantumRegister(2, "q")
+        c = ClassicalRegister(1, "c")
+        qc = QuantumCircuit(q, c, name="conditional")
+        qc.cx(q[0], q[1])
+
+        experiment = self._circuit_to_experiment(qc)
+        self.assertRaisesRegex(QiskitBackendError, 'Number of classical bits \(1\) is not sufficient for storing the '
+                                                   'outcomes of the experiment',
+                               self.measurements.collect_measurements, experiment)
+
+    def test_invalid_amount_of_classical_bits(self):
```

Comment:
ultra nit 😉 : for a countable thing like "bits", we would say "number of bits", rather than "amount of bits". "Amount" is typically only used when talking about an uncountable quantity.

Reply:
I will change (also in the exception raised)